src/realm/sync/client.cpp

```diff
@@ -1644,6 +1652,35 @@ util::Future<std::string> SessionWrapper::send_test_command(std::string body)
     return m_sess->send_test_command(std::move(body));
 }
 
+void SessionWrapper::handle_pending_client_reset_acknowledgement()
+{
+    auto pending_reset = [&] {
+        auto ft = m_db->start_frozen();
+        return _impl::client_reset::has_pending_reset(ft);
+    }();
+    REALM_ASSERT(pending_reset);
```

Comment:
Isn't it safer to just return early if there is no pending reset?

This flow feels racy because we check for pending reset state in a read transaction and then clear it on a separate write transaction. Are we allowed to assume this here because of the single threaded nature of the sync client?

Reply:
I kinda think it's equally safe to check pre-conditions vs returning early if pre-conditions aren't met?

Reply:
Hmm ok. Can we be certain that no other writer can clear the pending reset between when it gets written and when this frozen read transaction starts here?

Reply:
Yes, in the places we call this function we can guarantee it because we always call it after checking pre-conditions.